### DISCUSSION TITLE
Explicit padding explanation

### DISCUSSION THREAD

**Discussion by alexander-camuto** [sonos/tract#1143] [General]
Heya was wondering if there was a resource for explaining what the different parameters within 
```rust
Explicit(before, after, bool)
```
correspond to. 

It seems like the meaning of it varies per op. Eg. meaning will differ from Deconv vs Conv op. Is that correct ?

**Comment by kali** [comment]
Hello. Deconv is essentially a Conv upside-down. It sounds simple like that, but the details are an absolute brain twister.
The explicit padding will be apply on the non convoluted side of the operation. So it's equivalent to padding the convolution input, or cropping the deconv output...

You can look at https://github.com/sonos/tract/blob/main/core/src/ops/cnn/padding.rs where this logic is centralized (for conv, deconv, but also pooling operations), there are actually a couple of unit tests.

To add to this specific nightmare, onnx rules are super weird (I suspect there was a misplaced parens in the initial spec, but now everybody is reproducing it, so we kinda have to follow though too).

**Reply by alexander-camuto** [reply]
ty for your reply !
